packages/flame/test/widgets/nine_tile_box_widget_test.dart

```diff
@@ -0,0 +1,34 @@
+import 'package:flame/widgets.dart';
+import 'package:flame_test/flame_test.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+void main() async {
+  final image = await generateImage();
+
+  group('NineTileBoxWidget', () {
+    testWidgets('has no FutureBuilder when pass animation', (tester) async {
+      await tester.pumpWidget(
+        NineTileBoxWidget(
+          image: image,
+          tileSize: 1,
+          destTileSize: 1,
+        ),
+      );
+
+      final futureBuilderFinder = find.byType(FutureBuilder);
+      final nineTileBoxWidgetFinder = find.byType(NineTileBoxWidget);
+
+      expect(futureBuilderFinder, findsNothing);
+      expect(nineTileBoxWidgetFinder, findsOneWidget);
+    });
+
+    testWidgets(
+      'has FutureBuilder when pass asset path',
+      (tester) async {
+        ///How can I test this...?
```

Comment:
Maybe you can expose it with `@visibleForTesting`?

Reply:
I mean that how can I mock or inject the image for the `Constructor.asset` because it loads image internally but I found something.
Thank you :)